cf-custom-resources/lib/dns-cert-validator.js

```diff
@@ -164,30 +188,52 @@ const requestCertificate = async function (
   }
   if (attempt === maxAttempts) {
     throw new Error(
-      `DescribeCertificate did not contain DomainValidationOptions after ${maxAttempts} tries.`
+        `DescribeCertificate did not contain DomainValidationOptions after ${maxAttempts} tries.`
     );
   }
+  return options;
+};
 
-  await updateHostedZoneRecords(
-    "UPSERT",
+/**
+ * Validate DNS in the root, app, and env hosted zones in parallel, and wait until the certificate is validated.
+ * The root hosted zone is created when the user purchases example.com in route53 in their app account.
+ * We create the app hosted zone "app.example.com" when running "app init" part of the application stack.
+ * The env hosted zone "env.app.example.com" is created when running "env init" part of the env stack.
+ *
+ * @param options
+ * @param envRoute53
+ * @param appRoute53
+ * @param envHostedZoneId
+ * @param certificateARN
+ * @param acm
+ * @returns {Promise<>} promise for whether the wait succeeds
+ */
+const validateCertificate = async function(
     options,
     envRoute53,
     appRoute53,
-    envHostedZoneId
+    envHostedZoneId,
+    certificateARN,
+    acm
+){
+  await updateHostedZoneRecords(
+      "UPSERT",
+      options,
+      envRoute53,
+      appRoute53,
+      envHostedZoneId
   );
 
   await acm
     .waitFor("certificateValidated", {
       // Wait up to 9 minutes and 30 seconds
       $waiter: {
-        delay: 30,
-        maxAttempts: 19,
+        delay: 1,
+        maxAttempts: 1,
```

Comment:
Do we want these changes?

Reply:
oh no.. good catch and sorry :woman-facepalming: